pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -87,20 +97,79 @@ func (c *APIClient) connect() error {
 	return err
 }
 
-func parseKubeConfig(fpath string) (*k8s.Config, error) {
-	yamlFile, err := ioutil.ReadFile(fpath)
+// ParseKubeConfig reads and unmarcshals a kubeconfig file
+// in an object ready to use. Exported for integration testing.
+func ParseKubeConfig(fpath string) (*k8s.Config, error) {
+	// TODO: support yaml too
+	jsonFile, err := ioutil.ReadFile(fpath)
 	if err != nil {
 		return nil, err
 	}
 
 	config := &k8s.Config{}
-	err = yaml.Unmarshal(yamlFile, config)
+	err = json.Unmarshal(jsonFile, config)
 	return config, err
 }
 
+// ComponentStatuses returns the component status list from the APIServer
 func (c *APIClient) ComponentStatuses() (*v1.ComponentStatusList, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
 	defer cancel()
 	return c.client.CoreV1().ListComponentStatuses(ctx)
+}
+
+// ConfigMapTokenFetcher returns the value of the `tokenValue` from the `tokenKey` in the ConfigMap `configmaptokendca` if its timestamp is less than tokenTimeout old.
+func (c *APIClient) ConfigMapTokenFetcher(tokenKey string) (string, bool, error) {
```

Comment:
Let's make a backlog card about it: I'd prefer to get a `ConfigMap` object from the apiclient and call it's `Get`/`Set` methods, like the `DockerListener` creates `Services` that provide methods.

Later we'll be able to derive an interface other orchestrators can provide for value persistence.

- For now, just remove references to tokens, we store/retrieve a value in a configmap name

- The name is not great, if the suffix is a substantive (`Fetcher`) we usually assume it's an object/class, method names usually are verbs